routes/user.js

```diff
@@ -0,0 +1,77 @@
+var express = require('express');
+var router = express.Router();
+var User = require('../models/index.js').User;
+var mail = require('../config/nodeMailer');
+
+router.get('/:username', isLoggedIn, function(req, res, next) {
+  var user = req.user;
+  res.render('profile', { title: user.username + "'s Profile", user: user});
+});
+
+router.get('/:username/edit', isLoggedIn, function(req, res, next) {
+  var user = req.user;
+  res.render('editProfile', { title: 'Edit ' + user.username + "'s Profile", user: user});
+});
+
+
+router.post('/:username/update', isLoggedIn, function(req, res, next) {
```

Comment:
The update in the route is more explicit but you probably don't need it. Since it's a post and you pass it an existing user I can tell by the signature.